src/main/java/cn/nukkit/entity/item/EntityBoat.java

```diff
@@ -208,11 +208,11 @@ public boolean onUpdate(int currentTick) {
 
     @Override
     public boolean onInteract(Player player, Item item) {
-        if (this.linkedEntity != null) {
+        if (this.linkedEntity.length() > 2) {
```

Comment:
Its should be `if (this.linkedEntity.length() < 2) {`

Reply:
this not return false ? return false = cannot enter

Reply:
It should return always false